ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -64,9 +64,13 @@ - (void)applyChangeset:(CKTransactionalComponentDataSourceChangeset *)changeset
                               userInfo:userInfo];
 }
 
-static void applyChangesToCollectionView(CKTransactionalComponentDataSourceAppliedChanges *changes, UICollectionView *collectionView)
+static void applyChangesToCollectionView(CKTransactionalComponentDataSourceAppliedChanges *changes, UICollectionView *collectionView, CKTransactionalComponentDataSourceState *currentState, CKComponentDataSourceAttachController *attachController)
 {
-  [collectionView reloadItemsAtIndexPaths:[changes.updatedIndexPaths allObjects]];
+  [changes.updatedIndexPaths enumerateObjectsUsingBlock:^(NSIndexPath *indexPath, BOOL *stop) {
+	if (CKCollectionViewDataSourceCell *cell = (CKCollectionViewDataSourceCell*) [collectionView cellForItemAtIndexPath:indexPath]) {
+		_attachToCell(cell, indexPath, currentState, attachController);
+	}
+  }];
```

Comment:
Ok, provided I haven't misunderstood @benlodotcom's comment, there's actually no need for the mutable array at all, or doing a reload on any cells that aren't currently visible.